draft-ietf-httpbis-resumable-upload.md

```diff
@@ -237,34 +238,34 @@ upload-offset: 100
 :scheme: https
 :authority: example.com
 :path: /upload
-upload-token: :SGVs…SGU=:
-upload-draft-interop-version: 2
+upload-draft-interop-version: 3
 upload-incomplete: ?1
 content-length: 25
 [partial content (25 bytes)]
 
 :status: 201
+location: https://example.com/upload/b530ce8ff
 upload-incomplete: ?1
 upload-offset: 25
 ~~~
 
-The client MAY automatically attempt upload resumption when the connection is terminated unexpectedly, or if a server error status code between 500 and 599 (inclusive) is received. The client SHOULD NOT automatically retry if a client error status code between 400 and 499 (inclusive) is received.
+If the client received an informational repsonse with the upload URL, it MAY automatically attempt upload resumption when the connection is terminated unexpectedly, or if a server error status code between 500 and 599 (inclusive) is received. The client SHOULD NOT automatically retry if a client error status code between 400 and 499 (inclusive) is received.
 
 File metadata can affect how servers might act on the uploaded file. Clients can send Representation Metadata (see {{Section 8.3 of HTTP}}) in the Upload Creation Procedure request that starts an upload. Servers MAY interpret this metadata or MAY ignore it. The `Content-Type` header can be used to indicate the MIME type of the file. The `Content-Disposition` header can be used to transmit a filename. If included, the parameters SHOULD be either `filename`, `filename*` or `boundary`.
 
 ## Feature Detection
 
-If the client has no knowledge of whether the server supports resumable upload, the Upload Creation Procedure MAY be used with some additional constraints. In particular, the `Upload-Incomplete` header field ({{upload-incomplete}}) MUST NOT be sent in the request if the server support is unclear. This allows the upload to function as if it is a regular upload.
+If the client has no knowledge of whether the server supports resumable upload, the Upload Creation Procedure MAY be used with some additional constraints. In particular, the `Upload-Incomplete` header field ({{upload-incomplete}}) MUST NOT be set to true if the server support is unclear. This allows the upload to function as if it is a regular upload.
 
-If the server detects the Upload Creation Procedure and it supports resumable upload, an informational response with `104 (Upload Resumption Supported)` status MAY be sent to the client while the request body is being uploaded.
+The server SHOULD send the `104 (Upload Resumption Supported)` informational response to the client, to indicate its support for a resumable upload.
```

Comment:
Is `SHOULD` too strong for feature detection? I used `MAY` since it's optional, but we could update it all to `SHOULD`.

Reply:
Personally, I think SHOULD is better here. Yes, it is optional, but I do think that there is a benefit to using it and hence we should recommend it instead of just providing this mechanism.